src/main/java/org/openlmis/integration/dhis2/web/element/DhisElementComboController.java

```diff
@@ -0,0 +1,99 @@
+import org.openlmis.integration.dhis2.exception.NotFoundException;
+import org.openlmis.integration.dhis2.i18n.MessageKeys;
+import org.openlmis.integration.dhis2.repository.dataset.DatasetRepository;
+import org.openlmis.integration.dhis2.service.DhisDataService;
+import org.openlmis.integration.dhis2.util.Pagination;
+import org.openlmis.integration.dhis2.web.BaseController;
+import org.openlmis.integration.dhis2.web.dataset.DatasetController;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Controller;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+/**
+ * Controller used to expose concatenated DHIS2 Data Elements with Category Option Combos via HTTP.
+ */
+@Controller
+@RequestMapping(DhisElementComboController.RESOURCE_PATH)
+@Transactional
+public class DhisElementComboController extends BaseController {
+
+  public static final String RESOURCE_PATH = DatasetController.RESOURCE_PATH
+      + "/{datasetId}/elementsAndCombos";
+
+  @Autowired
+  private DatasetRepository datasetRepository;
+
+  @Autowired
+  private DhisDataService dhisDataService;
+
+  /**
+   * Retrieves all dhis element and category option combos combinations for a given server.
+   */
+  @GetMapping
+  @ResponseStatus(HttpStatus.OK)
+  @ResponseBody
+  public Page<DhisElementCombo> getElementsAndCombos(
+      @PathVariable("serverId") UUID serverId, @PathVariable("datasetId") UUID datasetId,
+      Pageable pageable) {
+    Dataset dataset = datasetRepository.findById(datasetId)
+        .orElseThrow(() -> new NotFoundException(MessageKeys.ERROR_DATASET_NOT_FOUND));
+    Server server = dataset.getServer();
+
+    String dhisDatasetId = dataset.getDhisDatasetId();
+    DhisDataset dhisDataset = dhisDataService.getDhisDataSetById(dhisDatasetId, server.getUrl(),
+        server.getUsername(), server.getPassword());
+    List<DhisDataElement> dhisDataElements = dhisDataset.getDataSetElements();
+
+    List<DhisCategoryOptionCombo> categoryOptionCombos = dhisDataService
+        .getDhisCategoryOptionCombos(server.getUrl(), server.getUsername(), server.getPassword());
+
+    List<DhisElementCombo> dhisElementCombos = new ArrayList<>();
+    for (DhisCategoryOptionCombo combo: categoryOptionCombos) {
+      for (DhisDataElement element: dhisDataElements) {
+        DhisElementCombo dhisElementCombo = new DhisElementCombo(
+            element.getName() + " - " + combo.getDisplayName(),
+            element.getName(),
+            combo.getDisplayName()
```

Comment:
to concat string please use StringBuilder or some StringUtils methods

Reply:
used StringBuilder here, as adviced